src/client/Microsoft.Identity.Client/Platforms/uap/WamBroker/AccountPicker.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Instance;
+using Windows.Security.Authentication.Web.Core;
+using Windows.Security.Credentials;
+using Windows.UI.ApplicationSettings;
+
+namespace Microsoft.Identity.Client.Platforms.uap.WamBroker
+{
+    internal class AccountPicker : IAccountPicker
+    {
+        private readonly IntPtr _parentHandle;
+        private readonly ILoggerAdapter _logger;
+        private readonly SynchronizationContext _synchronizationContext;
+        private readonly Authority _authority;
+        private readonly bool _isMsaPassthrough;
+        private readonly string _optionalHeaderText;
+        private volatile WebAccountProvider _provider;
+
+        public AccountPicker(
+            IntPtr parentHandle,
+            ILoggerAdapter logger,
+            SynchronizationContext synchronizationContext,
+            Authority authority,
+            bool isMsaPassthrough,
+            string optionalHeaderText)
+        {
+            _parentHandle = parentHandle;
+            _logger = logger;
+            _synchronizationContext = synchronizationContext;
+            _authority = authority;
+            _isMsaPassthrough = isMsaPassthrough;
+            _optionalHeaderText = optionalHeaderText;
+            _logger.Verbose(()=>"Is MSA passthrough? " + _isMsaPassthrough);
+        }
+
+        public async Task<WebAccountProvider> DetermineAccountInteractivelyAsync()
+        {
+            await ShowPicker_UWPAsync().ConfigureAwait(true);
```

Comment:
UAP / UWP, we should stick on to one. But if this old code leave it as is

Reply:
Old code.